nuxeo-services/nuxeo-platform-filemanager-core/src/test/java/org/nuxeo/ecm/platform/filemanager/TestCSVZipImporter.java

```diff
@@ -0,0 +1,197 @@
+
+    protected DocumentModel wsRoot;
+
+    private static File getArchiveFile(String file) {
+        return new File(FileUtils.getResourcePathFromContext(file));
+    }
+
+    @Before
+    public void createTestDocuments() throws Exception {
+        wsRoot = coreSession.getDocument(new PathRef("/default-domain/workspaces"));
+
+        workspace1= coreSession.createDocumentModel(wsRoot.getPathAsString(), "ws1", "Workspace");
+        workspace1.setProperty("dublincore", "title", "test WS1");
+        workspace1 = coreSession.createDocument(workspace1);
+
+        workspace2= coreSession.createDocumentModel(wsRoot.getPathAsString(), "ws2", "Workspace");
+        workspace2.setProperty("dublincore", "title", "test WS2");
+        workspace2 = coreSession.createDocument(workspace2);
+    }
+
+    @Test
+    public void testArchiveDetection() throws Exception {
+        ZipFile archive = CSVZipImporter.getArchiveFileIfValid(getArchiveFile("test-data/testCSVArchive.zip"));
+        assertNotNull(archive);
+        archive.close();
+    }
+
+    @Test
+    public void testImportViaFileManager() throws Exception {
+        File archive = getArchiveFile("test-data/testCSVArchive.zip");
+        FileManager fm = Framework.getService(FileManager.class);
+        Blob blob = Blobs.createBlob(archive);
+        FileImporterContext context = FileImporterContext.builder(coreSession, blob, workspace1.getPathAsString())
+                                                         .overwrite(true)
+                                                         .build();
+        fm.createOrUpdateDocument(context);
+        DocumentModelList children = coreSession.getChildren(workspace1.getRef());
+        assertSame(2, children.size());
+
+        DocumentModel MyFile = coreSession.getChild(workspace1.getRef(), "MyFile");
+        DocumentModel MyNote = coreSession.getChild(workspace1.getRef(), "MyNote");
+
+        assertEquals("My File", MyFile.getTitle());
+        assertEquals("My Note", MyNote.getTitle());
+
+        assertEquals("MyFile", MyFile.getName());
+        assertEquals("MyNote", MyNote.getName());
+
+        assertEquals("this is text", MyNote.getProperty("note", "note"));
+        assertNotNull(MyFile.getProperty("file", "content"));
+    }
+
+    @Test
+    public void testDocumentCreationFailureWithNoName() throws Exception {
+        File archive = getArchiveFile("test-data/testNoNameCSVArchive.zip");
+        FileManager fm = Framework.getService(FileManager.class);
+        Blob blob = Blobs.createBlob(archive);
+        FileImporterContext context = FileImporterContext.builder(coreSession, blob, workspace2.getPathAsString())
+                                                         .overwrite(true)
+                                                         .build();
+        fm.createOrUpdateDocument(context);
+        DocumentModelList children = coreSession.getChildren(workspace2.getRef());
+        assertSame(0, children.size());
```

Comment:
assertEquals everytime you compare two numbers that fit in a long

Reply:
updated to assertEquals